src/internal/modules/billing/controllers/bill-run.js

```diff
@@ -258,32 +253,69 @@ const getBillingBatchEmpty = async (request, h) => {
 }
 
 /**
- * Deletes all billing data
+ * Renders an error page if the batch is errored - i.e. the charging module is down
+ * @param {Object} request.pre.batch - billing batch loaded by pre handler
  */
-const postDeleteAllBillingData = async (request, h) => {
-  await services.water.billingBatches.deleteAllBillingData()
-  return h.redirect(BATCH_LIST_ROUTE)
-}
-
-exports.getBillingBatchList = getBillingBatchList
-exports.getBillingBatchSummary = getBillingBatchSummary
-exports.getBillingBatchInvoice = getBillingBatchInvoice
+const getBillingBatchError = async (request, h) => {
+  const { batch } = request.pre
 
-exports.getBillingBatchCancel = getBillingBatchCancel
-exports.postBillingBatchCancel = postBillingBatchCancel
+  return h.view('nunjucks/billing/batch-error', {
+    ...request.view,
+    pageTitle: getBillRunPageTitle(batch),
+    batch,
+    back: BATCH_LIST_ROUTE,
+    errorList: _errorList(batch.errorCode)
+  })
+}
 
-exports.getBillingBatchConfirm = getBillingBatchConfirm
-exports.postBillingBatchConfirm = postBillingBatchConfirm
-exports.getBillingBatchConfirmSuccess = getBillingBatchConfirmSuccess
+const _errorList = (errorCode) => {
+  if (!errorCode) {
+    return [{ text: 'No error code was assigned. We have no further information at this time.' }]
+  }
 
-exports.getBillingBatchDeleteInvoice = getBillingBatchDeleteInvoice
-exports.postBillingBatchDeleteInvoice = postBillingBatchDeleteInvoice
+  const errors = [
+    { code: 10, text: 'Error when populating the charge versions.' },
+    { code: 20, text: 'Error when processing the charge versions.' },
+    { code: 30, text: 'Error when preparing the transactions.' },
+    { code: 40, text: 'Error when requesting or processing a transaction charge.' },
+    { code: 50, text: 'Error when creating the Charging Module bill run.' },
+    { code: 60, text: 'Error when deleting an invoice.' },
+    { code: 70, text: 'Error when processing two-part tariff.' },
+    { code: 80, text: 'Error when getting the Charging Module bill run summary.' },
+    { code: 90, text: 'Error when re-billing a bill run.' }
+  ]
+
+  const error = errors.find((error) => {
+    return error.code === errorCode
+  })
```

Comment:
Just spotted your comment from last week on coding conventions and realised this is now a truthiness check we could condense to one line!
```suggestion
  const error = errors.find((error) => error.code === errorCode)
```

Reply:
In my head, the exception related specifically to where you are just asserting a value is 'truthy'. i.e. it's not `null` or `undefined`.

I would still use _block_ body for the check we're doing here because I consider it an equality check rather than truth.

So, I will take your approval and run with it!